### PR TITLE
Make `xcm-emulator` test pass for new consensus hook

### PR DESCRIPTION
**Pull request by NachoPal**
This PR tries to solve: https://github.com/paritytech/cumulus/issues/2703

Tests are not panicking anymore with `slot info is inserted on block initialization`, but now they are running into another issue. Tests are failing with `included head not present in relay storage proof`.

Should I add the rest of the hooks (`on_finalize`, etc)?

### CODE REVIEW

**Comment by slumber** [comment]
We need a full parent header to be included into the storage proof. For example, this is how it's done for parachain system tests
https://github.com/paritytech/cumulus/blob/6141995fa6bcee862d2d0ffaefb641b52a8e3b6c/pallets/parachain-system/src/tests.rs#L403-L404

https://github.com/paritytech/cumulus/blob/6141995fa6bcee862d2d0ffaefb641b52a8e3b6c/pallets/parachain-system/src/tests.rs#L347-L351

To my understanding, headers are not accessible from the runtime, so we need system pallet to be finalized.

**Comment by muharem** [comment]
I would say that the emulator should behave as close as possible to the real environment, while only mocking the network layer. `on_finalize` should also run for all pallets, in a right order.

File: parachains/integration-tests/emulated/common/src/lib.rs [hunk 1]
```diff
@@ -27,6 +27,7 @@ decl_test_relay_chains! {
 			MessageQueue: polkadot_runtime::MessageQueue,
 			XcmConfig: polkadot_runtime::xcm_config::XcmConfig,
 			SovereignAccountOf: polkadot_runtime::xcm_config::SovereignAccountOf,
+			AllPallets: polkadot_runtime::AllPalletsWithSystem,
 			System: polkadot_runtime::System,
 			Balances: polkadot_runtime::Balances,
```
Comment (muharem) [inline parachains/integration-tests/emulated/common/src/lib.rs:32]: not sure if we even need to list the pallets, and pallets_extra. 
may be optionally, if there is some use case for it.

File: xcm/xcm-emulator/src/lib.rs [hunk 2]
```diff
@@ -553,16 +563,26 @@ macro_rules! __impl_test_ext_for_parachain {
...
+					})
+				});
 
 				let para_id = <$name>::para_id().into();
 
 				$ext_name.with(|v| {
 					v.borrow_mut().execute_with(|| {
 						// Make sure it has been recorded properly
-						let relay_block_number = <$name>::relay_block_number();
+						// let relay_block_number = <$name>::relay_block_number();
```
Comment (muharem) [inline xcm/xcm-emulator/src/lib.rs:585]:
```suggestion
```

**Comment by NachoPal** [comment]
bot fmt

**Comment by NachoPal** [comment]
I dropped the idea of running all pallets `on_initialized` and `on_finalized` with `AllPalletsWithSystem`. The reason is I was running into more errors because of multiple reasons. One of them was the slot verification when using defaults pre-runtime digest. It forced me to manually build them + having to set inherent for timestamp etc...

I realised I was building something very specific for our runtimes using `Aura` and `AuraExt`. One of the goals of `xcm-emulator` is to be as generic as possible, so other parachains (that are maybe using another consensus system) can use it too.

I created this new issue https://github.com/paritytech/cumulus/issues/2734 to add `on_initialize` and `on_finalize` optional hooks to `xcm-emulator`.

The changes in this PR should be good enough to make the test pass, and therefore not being a blocker anymore.

Please merge if you are ok with it.

**Review by gilescope — APPROVED** [review]
It's a closer emulation than what we've currently got so it gets my vote.

**Review by slumber — APPROVED** [review]

**Comment by slumber** [comment]
Thanks @NachoPal!

**Pull request merged**